pkg/aws/helpers.go

```diff
@@ -457,8 +484,80 @@ func GetPrefixFromOperatorRole(cluster *cmv1.Cluster) string {
 	return rolePrefix
 }
 
-func GetAccountRoleName(cluster *cmv1.Cluster) (string, error) {
-	return GetResourceIdFromARN(cluster.AWS().STS().RoleARN())
+func GetOperatorRolePolicyPrefixFromCluster(cluster *cmv1.Cluster, awsClient Client) (string, error) {
+	installerRolePrefix, err := GetPrefixFromInstallerAccRole(cluster)
+	if err != nil {
+		return "", err
+	}
+	// Check if installer role prefix follows standard
+	installerRoleName, err := GetResourceIdFromARN(cluster.AWS().STS().RoleARN())
+	if err != nil {
+		return "", err
+	}
+	hasStandardAccRole := installerRolePrefix != installerRoleName
```

Comment:
Non standard roles, if the role does not have suffix Installer-Role, Worker-Role etc. Then after getting the prefix it will still be the whole name since we check if it HasSuffix to get the prefix.